### PR TITLE
[MAINTENANCE] Minor cleanup for better code readability

### PR DESCRIPTION
**Pull request by alexsherstinsky**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-1347/GREAT-181/GREAT-1423
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/validator/validator.py [hunk 1]
```diff
@@ -1140,13 +1144,16 @@ def _generate_metric_dependency_subgraphs_for_each_expectation_configuration(
...
 
-        return evrs, processed_configurations
+        return expectation_validation_graphs, evrs, processed_configurations
 
     def _generate_suite_level_graph_from_expectation_level_sub_graphs(
         self,
         expectation_validation_graphs: List[ExpectationValidationGraph],
     ) -> ValidationGraph:
         # Collect edges from all expectation-level sub-graphs and incorporate them under common suite-level graph.
+        if expectation_validation_graphs is None:
```
Comment (NathanFarmer) [inline great_expectations/validator/validator.py]: According to the method signature this cannot be `None`.

File: great_expectations/validator/validator.py [hunk 2]
```diff
@@ -1186,6 +1193,9 @@ def _resolve_suite_level_graph_and_process_metric_evaluation_errors(
         )
 
         # Trace MetricResolutionError occurrences to expectations relying on corresponding malfunctioning metrics.
+        if expectation_validation_graphs is None:
```
Comment (NathanFarmer) [inline great_expectations/validator/validator.py]: Same comment here

**Review by NathanFarmer — CHANGES_REQUESTED** [review]
I would expect this to fail the static type checking step.

File: great_expectations/rule_based_profiler/helpers/simple_semantic_type_filter.py [hunk 3]
```diff
@@ -68,19 +70,20 @@ def parse_semantic_domain_type_argument(
                     SemanticDomainTypes(semantic_type)
                     for semantic_type in semantic_types
                 ]
-            elif all(
+
+            if all(
                 [
                     isinstance(semantic_type, SemanticDomainTypes)
                     for semantic_type in semantic_types
                 ]
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/helpers/simple_semantic_type_filter.py]: Nit - we don't need to create a list here. `all` works with any iterator so the extra memory is unnecessary

File: great_expectations/validator/validator.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/validator/validator.py]: Incorporated -- thanks!

File: great_expectations/validator/validator.py (hunk 2 above)
Comment (alexsherstinsky) [reply great_expectations/validator/validator.py]: @NathanFarmer Incorporated -- thanks!

File: great_expectations/rule_based_profiler/helpers/simple_semantic_type_filter.py (hunk 3 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/helpers/simple_semantic_type_filter.py]: @cdkini -- Incorporated -- thanks.

**Review by NathanFarmer — APPROVED** [review]
LGTM!

**Pull request merged**